src/Orleans/Streams/Internal/StreamConsumerExtension.cs

```diff
@@ -5,13 +5,14 @@
 using System.Threading.Tasks;
 using Orleans.Concurrency;
 using Orleans.Runtime;
+using Orleans.Serialization;
 
 namespace Orleans.Streams
 {
     internal interface IStreamSubscriptionHandle
     {
         Task<StreamHandshakeToken> DeliverItem(object item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken);
-        Task<StreamHandshakeToken> DeliverBatch(IBatchContainer item, StreamHandshakeToken handshakeToken);
+        Task<StreamHandshakeToken> DeliverBatch(IBatchContainer item, StreamHandshakeToken handshakeToken, SerializationManager serializationManager);
```

Comment:
I'll take another look at it - I don't like it either, but it was discussed with Jason at the time. The core issue is that many of these types are `[Serializable]` and some of the important generic types have the `new()` constraint, eg:
```C#
public class MemoryStreamProvider<TSerializer> : PersistentStreamProvider<MemoryAdapterFactory<TSerializer>>
    where TSerializer : IMemoryMessageBodySerializer, new()
```